### ISSUE TITLE
It is currently in use by another Gradle instance.

### ISSUE THREAD

**Issue by ritesh-singh** [gradle/gradle#30135]
### Current Behavior

Running multiple daemons for the same build, leads to below error

```
Caused by: java.lang.RuntimeException: Failed to load cache entry 6a01857fda538e97a49cf02cfg40058b for Kotlin DSL version catalog plugin accessors for classpath '22113a1c8cdbe86a08b1b01087432360': Could not load from local cache: Timeout waiting to lock Build cache (/Users/.gradle/caches/build-cache-1). It is currently in use by another Gradle instance.
Owner PID: 89317
Our PID: 99001
Owner Operation: 
Our operation: 
Lock file: /Users/.gradle/caches/build-cache-1/build-cache-1.lock
```

### Expected Behavior

Should not lead to build failure, when multiple daemons are trying to access or modify the cache simultaneously from same location, there needs to be a way to handle concurrent access in a safe way without causing data corruption or any kind of inconsistency.

### Steps to Reproduce

Enable local cache
`org.gradle.caching=true`

Reproducing step
- Run a gradle build from CLI
- Same, time perform IDE sync from IDE (AndroidStudio/Intellij)

### Gradle version

8.6

**Comment by cookieMr** [comment]
I have similar issue (probably the same).
And the issue is not related to version that runs (we use a wide variety of versionst, starting from 8.5, iup to latest 8.9).

```bash
./gradlew clean build test check --rerun --no-build-cache
```
```
Starting a Gradle Daemon, 8 stopped Daemons could not be reused, use --status for details

FAILURE: Build failed with an exception.

* What went wrong:
Gradle could not start your build.
> Cannot create service of type BuildSessionActionExecutor using method LauncherServices$ToolingBuildSessionScopeServices.createActionExecutor() as there is a problem with parameter #21 of type BuildLifecycleAwareVirtualFileSystem.
   > Cannot create service of type BuildLifecycleAwareVirtualFileSystem using method VirtualFileSystemServices$GradleUserHomeServices.createVirtualFileSystem() as there is a problem with parameter #1 of type FileWatchingFilter.
      > Cannot create service of type FileWatchingFilter using method VirtualFileSystemServices$GradleUserHomeServices.createFileWatchingFilter() as there is a problem with parameter #1 of type GlobalCacheLocations.
         > Cannot create service of type GlobalCacheLocations using method GradleUserHomeScopeServices.createGlobalCacheLocations() as there is a problem with parameter #1 of type List<GlobalCache>.
            > Could not create service of type FileAccessTimeJournal using GradleUserHomeScopeServices.createFileAccessTimeJournal().
               > Failed to ping owner of lock for journal cache (/Users/cookie/.gradle/caches/journal-1) (lock id: 5868202984269084883, port: 64760)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 2s
```

```bash
```
```
   PID STATUS   INFO
 58392 IDLE     8.9
 40629 STOPPED  (stop command received)

```

**Comment by ov7a** [comment]
This issue needs a decision from the team responsible for that area. They have been informed. Response time may vary.

***
There are still some parts of Gradle that require exclusive access. The use case about running two builds in parallel (IDE and console) is valid and ideally, Gradle should not fail. Even if it fails, the error should be more descriptive.

**Comment by cookieMr** [comment]
> This issue needs a decision from the team responsible for that area. They have been informed. Response time may vary.
> 
> There are still some parts of Gradle that require exclusive access. The use case about running two builds in parallel (IDE and console) is valid and ideally, Gradle should not fail. Even if it fails, the error should be more descriptive.

Hi, thank you for your reply. I truly appreciate that this issue is not slowly sinking in the backlog/issue tracker.

As for the parallelism... yes, this is the ideal case. However, the issue (for me) is also, that when i run just 1 gradle daemon, it blocks 2nd run, even the same cmd in the same terminal window, even when the 1st daemon finished work. The 2nd daemon is complaining about not being able `to ping owner of lock for file`.

Random question: maybe hash cache is corrupted?

**Comment by tomekrojek** [comment]
Got similar error when running multiple gradle builds in parallel in a project that has convention plugins kept as composite build. I have daemon disabled but there are multiple gradle executions in parallel at the same time.

**Comment by ov7a** [comment]
FTR, If anyone can consistently reproduce the issue and can provide a self-contained reproducer project, it would help to investigate this.

**Comment by cookieMr** [comment]
It seems the issue is not within this repo.

I am unable to reproduce this issue on my Ubuntu laptop.
But i was able to reproduce this on MacOS (13.6.9) before the update.
Not on updated MacOS (14.7) the issue is gone. I am running (like a crazy monster about cookies) different gradle tasks/projects/jdk versions/gradle versions and i am not able to reproduce this issue after the update.

@ritesh-singh @tomekrojek can you confirm this observation of mine?

**Comment by ritesh-singh** [comment]
Unless, this error is fixed in further versions - i don't expect it to be fixed. It should not be dependent on OS IMO.

**Comment by castelblanque-broadcom** [comment]
I am consistently suffering this issue in my Intel Mac.
Latest version of IntelliJ, stopping all daemons always.

Is there any update on this?

**Comment by phillipgziprecruiter** [comment]
I'm also affected by this consistently:
------------------------------------------------------------
Gradle 8.8
------------------------------------------------------------

Build time:   2024-05-31 21:46:56 UTC
Revision:     4bd1b3d3fc3f31db5a26eecb416a165b8cc36082

Kotlin:       1.9.22
Groovy:       3.0.21
Ant:          Apache Ant(TM) version 1.10.13 compiled on January 4 2023
JVM:          17.0.8 (Amazon.com Inc. 17.0.8+7-LTS)
OS:           Mac OS X 15.1 aarch64

**Comment by ov7a** [comment]
Related:
* https://github.com/gradle/gradle/issues/8750

**Comment by enggalilio** [comment]
We have the same issue while running multiple builds in our CI environment.

Is there any plan to fix it in the future as all workarounds are overwhelming and not consistent

**Comment by BoD** [comment]
```
Gradle could not start your build.
> Could not create service of type BuildLifecycleController using ServicesProvider.createBuildLifecycleController().
   > Could not create service of type BuildModelController using VintageBuildControllerProvider.createBuildModelController().
      > Could not create service of type FileHasher using BuildSessionServices.createFileHasher().
         > Failed to ping owner of lock for file hash cache (/Users/bod/gitrepo/apollo-kotlin-0/.gradle/8.11/fileHashes) (lock id: -5953003056333596191, port: 65183)
```

I've been getting this often for the last few days. 

I updated my system to MacOS 15.2 recently - but I'm not 100% positive this 2 things are related as I can't remember for sure when I got the issue the first time.

Unfortunately I haven't found a way to repro, it seems random, but happening often. 

`./gradlew --stop` fixes it until next time.

**Comment by shalva97** [comment]
similar issue here

> Timeout waiting to lock file content cache (/Users/Giorgi_Shalvashvili/.gradle/caches/8.11.1/fileContent). It is currently in use by another Gradle instance.

**Comment by shalva97** [comment]
Here is my small .kts workaround.

```
// TODO remove in future
// workaround for .lock files preventing the app to be built
tasks.configureEach {
    if (name != "deleteLockFiles") {
        dependsOn("deleteLockFiles")
    }
}

tasks.register("deleteLockFiles") {
    doLast {
        // List of specific lock file paths to delete
        val lockFilePaths = listOf(
            "~/.gradle/caches/8.11.1/fileContent/fileContent.lock",
            "~/.gradle/caches/modules-2/modules-2.lock",
            "~/.gradle/caches/8.11.1/generated-gradle-jars/generated-gradle-jars.lock",
            "~/.gradle/caches/journal-1/journal-1.lock"
        )

        // Delete lock files
        lockFilePaths.forEach { path ->
            val file = project.file(path)
            if (file.exists()) {
                println("Deleting lock file: ${file.absolutePath}")
                file.delete()
            }
        }
    }
}
```

**Comment by GabeK0** [comment]
Wanted to flag this somewhat related issue: https://github.com/gradle/gradle/issues/31245

People experiencing these timeouts more recently (around December 2024) could have the root cause be from firewall software + macOS 15, and disabling said firewall software can be a potential temporary workaround. That might be the fix for you @BoD @shalva97

**Comment by mpermar** [comment]
I think the team is already on track on fixing this as per @GabeK0's comment above  but just came to say that this issue is easily reproducible if you happen to have Symantec WSS agent installed in your machine. You won't be able to run two gradle daemons simultaneously with that agent turned on due to the reported lock access errors.

**Comment by caot** [comment]
* Add -Djava.io.tmpdir=$HOME/tmp  to fix "Cannot create service of type BuildSessionActionExecutor ...". e.g

mkdir -p $HOME/tmp
gradle clean -Djava.io.tmpdir=$HOME/tmp

**Comment by mlopatkin** [comment]
This issue will be closed as a duplicate of

* #8375

Please add your use case and 👍 to that issue.

If you think our analysis is wrong, please provide us with more detailed information explaining why.
